packages/medusa/src/services/totals.ts

```diff
@@ -1016,18 +1018,31 @@ class TotalsService extends TransactionBaseService {
     )
     const orderGiftCardAmount = Math.min(giftCardable, toReturn)
 
-    if (cartOrOrder.region?.gift_cards_taxable) {
-      return {
-        total: orderGiftCardAmount,
-        tax_total: Math.round(
-          (orderGiftCardAmount * cartOrOrder.region.tax_rate) / 100
-        ),
+    // If a gift card is not taxable, the tax_rate for the giftcard will be null
+    const { totalGiftCardBalance, totalTaxFromGiftCards } = cartOrOrder.gift_cards.reduce((acc, giftCard) => {
+      let taxableAmount = 0
+
+      acc.totalGiftCardBalance += giftCard.balance
+
+      // Add to the taxable amount only if the gift cards can be taxed.
+      if (cartOrOrder.region.gift_cards_taxable) {
```

Comment:
question: Is this still a valid check to be made? Since the tax_rate is already present in the gift card.

Reply:
It depends on whether we calculate the total for an order or a cart. 

When we calculate for an order, we have the `gift_card_transactions`, which will hold the `tax_rate` that the `gift_card` was made with. In this case, we don't need to take the `gift_cards_taxable` flag into account, as the settings at the time of purchase dictate the taxes.

When we calculate for a cart, we are computing what the total will become when the order is placed. So in this scenario, we should consider whether or not taxes are applied to gift cards in the region. 

The order gc total is calculated above using the `gift_card_transactions`, so the highlighted if-statement is in relation to cart calculations. I.e., the check should remain :)

Reply:
That is correct. But now that we have tax_rate on the gift card, we don't need to check this flag on a cart or an order for totals calculation. 

We use this flag to decide whether the `giftCard.tax_rate` should be filled with a tax_rate or not at the time of gift card creation. This, along with GiftCardTransaction, which already have the tax_rate will end up becoming the new source of truth for gift card taxation.

This would imply that us looking at this flag at totals calculation, would produce incorrect taxation data for that particular gift card. 

Correct me if i'm wrong here.

Reply:
Ah, I see your point! Wonder what the most correct approach would be here, but your suggestion of ignoring the current settings and just using the settings (taxable + tax rate) at the time of purchase.

Let's go with that!